src/USER-MISC/fix_pafi.cpp

```diff
@@ -0,0 +1,803 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+/* ------------------------------------------------------------------------
+   Contributing authors: Thomas Swinburne (CNRS & CINaM, Marseille, France)
+
+   Please cite the related publication:
+   T.D. Swinburne and M.-C. Marinica, Unsupervised calculation of free energy barriers in large crystalline systems, Physical Review Letters 2018
+------------------------------------------------------------------------- */
+
+
+#include "fix_pafi.h"
+#include "mpi.h"
+#include "math.h"
+#include "string.h"
+#include "stdlib.h"
+#include "math_extra.h"
+#include "atom.h"
+#include "force.h"
+#include "update.h"
+#include "modify.h"
+#include "domain.h"
+#include "region.h"
+#include "respa.h"
+#include "comm.h"
+#include "input.h"
+#include "variable.h"
+#include "compute.h"
+#include "random_mars.h"
+#include "memory.h"
+#include "error.h"
+#include "utils.h"
+#include "group.h"
+#include "citeme.h"
+
+using namespace LAMMPS_NS;
+
+static const char cite_user_pafi_package[] =
+  "citation for fix pafi:\n\n"
+  "@article{SwinburneMarinica2018,\n"
+  "author={T. D. Swinburne and M. C. Marinica},\n"
+  "title={Unsupervised calculation of free energy barriers in large crystalline systems},\n"
+  "journal={Physical Review Letters},\n"
+  "volume={276},\n"
+  "number={1},\n"
+  "pages={154--165},\n"
+  "year={2018},\n"
+  "publisher={APS}\n"
+  "}\n\n";
+
+using namespace FixConst;
+
+/* ---------------------------------------------------------------------- */
+
+FixPAFI::FixPAFI(LAMMPS *lmp, int narg, char **arg) :
+  Fix(lmp, narg, arg), idregion(NULL), random(NULL), computename(NULL),
+      h(NULL), step_respa(NULL)
+{
+  if (lmp->citeme) lmp->citeme->add(cite_user_pafi_package);
+
+  if (narg < 11) error->all(FLERR,"Illegal fix pafi command");
+
+  dynamic_group_allow = 0;
+  vector_flag = 1;
+  size_vector = 4;
+  global_freq = 1;
+  extvector = 0;
+  od_flag = 0;
+  com_flag = 0;
+  time_integrate = 1;
+
+  int n = strlen(arg[3])+1;
+  computename = new char[n];
+  strcpy(computename,&arg[3][0]);
+
+
+  icompute = modify->find_compute(computename);
+  char buffer[128];
+
+  if (icompute < 0) {
+    sprintf(buffer,"Compute %s for fix pafi does not exist",computename);
```

Comment:
please use `snprintf()` to guarantee that there is no crash due to a buffer overflow from somebody using very long compute-id.